tests/test_views.py

```diff
@@ -402,30 +413,37 @@ def test_get_course_specific_consent(
             enterprise_customer_user=ecu,
             course_id=course_id
         )
-        response = self.client.get(
+        url = (
             self.url + '?course_id=course-v1%3AedX%2BDemoX%2BDemo_Course&next=https%3A%2F%2Fgoogle.com'
         )
-        assert response.status_code == 200
-        for key, value in {
-                "platform_name": "My Platform",
-                "data_sharing_consent": "required",
-                "messages": {
-                    "note": (
-                        "Courses from Starfleet Academy require data sharing consent. If you do not agree to "
-                        "share your data, you will be redirected to your dashboard."
-                    ),
-                    "warning": (
-                        "Are you sure? If you do not agree to share your data "
-                        "with Starfleet Academy, you cannot access edX Demo Course."
-                    ),
-                },
-                "course_id": "course-v1:edX+DemoX+Demo_Course",
-                "course_name": "edX Demo Course",
-                "redirect_url": "https://google.com",
-                "enterprise_customer_name": ecu.enterprise_customer.name,
-                "course_specific": True
-        }.items():
-            assert response.context[key] == value  # pylint:disable=no-member
+        if enrollment_deferred:
+            url += '&enrollment_deferred=true'
+        if supply_customer_uuid:
+            url += '&enterprise_id={uuid}'.format(uuid=enterprise_customer.uuid)
```

Comment:
Ah, good idea. Yes, it's a request factory so this should work. I changed this up to use the data param.